modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/source/ExtensionMessageSource.java

```diff
@@ -309,7 +309,7 @@ public void onException(ConnectionException exception) {
                 exception);
 
     Mono<Void> reconnectionAction = sourceAdapter.getReconnectionAction(exception)
-        .map(p -> from(retryPolicyTemplate.applyPolicy(p, retryScheduler)))
+        .map(p -> from(retryPolicyTemplate.applyPolicy(p, retryScheduler)).doOnTerminate(() -> reconnecting.set(false)))
```

Comment:
where's the test case for this?

Reply:
I'm working on it. This PR is to validate that this fix does not break any tests

Reply:
This case is covered by: `org.mule.test.module.extension.source.CustomReconnectionSourceTestCase.successfulCustomReconnection`